internal/packages/status/status.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package status
+
+import (
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/packages/changelog"
+	"github.com/elastic/elastic-package/internal/registry"
+)
+
+// PackageStatus holds version and deployment information about a package
+type PackageStatus struct {
+	Name       string
+	Changelog  []changelog.Revision
+	Local      *packages.PackageManifest
+	Production []packages.PackageManifest
+	Staging    []packages.PackageManifest
+	Snapshot   []packages.PackageManifest
+}
+
+// LocalPackage returns the status of a given package including local development information
+func LocalPackage(packageRootPath string, showAll bool) (*PackageStatus, error) {
+	manifest, err := packages.ReadPackageManifestFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package manifest failed")
+	}
+	changelog, err := changelog.ReadChangelogFromPackageRoot(packageRootPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading package changelog failed")
+	}
+	status, err := Package(manifest.Name, showAll)
+	if err != nil {
+		return nil, err
+	}
+	status.Changelog = changelog
+	status.Local = manifest
+	return status, nil
+}
+
+// Package returns the status of a given package
+func Package(packageName string, showAll bool) (*PackageStatus, error) {
+	snapshotManifests, err := registry.Snapshot.Revisions(packageName, showAll)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving snapshot deployment failed")
+	}
+	stagingManifests, err := registry.Staging.Revisions(packageName, showAll)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving staging deployment failed")
+	}
+	productionManifests, err := registry.Production.Revisions(packageName, showAll)
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving production deployment failed")
+	}
+	return &PackageStatus{
+		Name:       packageName,
+		Snapshot:   snapshotManifests,
+		Staging:    stagingManifests,
+		Production: productionManifests,
+	}, nil
+}
+
+// PendingChanges returns pending changes in a local development package folder
+func (p *PackageStatus) PendingChanges() (*changelog.Revision, error) {
```

Comment:
Could you please explain why does `PendingChanges` method is actually a separate method :) ? I'm wondering if it can be just static part of the `PackageStatus`.